                                                                                                                                                                                                                                                               
Delivered-To: jwboyer@gmail.com
Received: by 10.76.6.212 with SMTP id d20csp344695oaa;
        Thu, 12 Jun 2014 08:28:19 -0700 (PDT)
X-Received: by 10.52.117.16 with SMTP id ka16mr607162vdb.72.1402586899259;
        Thu, 12 Jun 2014 08:28:19 -0700 (PDT)
Return-Path: <luto@amacapital.net>
Received: from bastion.fedoraproject.org (bastion02.fedoraproject.org. [209.132.181.3])
        by mx.google.com with ESMTP id av9si41982084pbd.28.2014.06.12.08.28.18
        for <jwboyer@gmail.com>;
        Thu, 12 Jun 2014 08:28:19 -0700 (PDT)
Received-SPF: softfail (google.com: domain of transitioning luto@amacapital.net does not designate 209.132.181.3 as permitted sender) client-ip=209.132.181.3;
Authentication-Results: mx.google.com;
       spf=softfail (google.com: domain of transitioning luto@amacapital.net does not designate 209.132.181.3 as permitted sender) smtp.mail=luto@amacapital.net
Received: by bastion02.phx2.fedoraproject.org (Postfix)
	id 99AB440A0C; Thu, 12 Jun 2014 15:28:18 +0000 (UTC)
Delivered-To: jwboyer@fedoraproject.org
Received: from mx1.redhat.com (ext-mx11.extmail.prod.ext.phx2.redhat.com [10.5.110.16])
	by bastion02.phx2.fedoraproject.org (Postfix) with ESMTP id 004DA409E2
	for <jwboyer@fedoraproject.org>; Thu, 12 Jun 2014 15:28:17 +0000 (UTC)
Received: from mail-pb0-f53.google.com (mail-pb0-f53.google.com [209.85.160.53])
	by mx1.redhat.com (8.14.4/8.14.4) with ESMTP id s5CFSGU5021153
	for <jwboyer@fedoraproject.org>; Thu, 12 Jun 2014 11:28:17 -0400
Received: by mail-pb0-f53.google.com with SMTP id uo5so852395pbc.40
        for <jwboyer@fedoraproject.org>; Thu, 12 Jun 2014 08:28:16 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20130820;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=ddu2G5u9JbFRc8+rX1oe08XVbsdCMg9zOAD5WpZAUW8=;
        b=fGKwrcdoPNjePGLV+Bc6LhnQhGwD8n1Ebib3P/R3WpBb0z4cnDERuW3gKfaytc6cOl
         xBP26iybb80BEyJFw7gJMDn7KATuyqTu5xc+XjsEJBnPAKxAQlupq97raiC6R10h0jeD
         XL4370OCs5p+GkiEBaH9PkNdbNB2f75RzBRvKwwgLCVQrQcY1KNDGsorxhkMEPLMCv6I
         tQM89Z4qlMI50Bu8lF6cPudAK2/SYHYfjVWPwZIbMRkbZamv0tcmTQZ860anhZJOq9wk
         1aRKO8ZU8sBfNeWsteXUGFYLo9hAqWzbpljGUjX0FGaq7ldT1jG34j5aZmemnsxRHY5U
         1jLA==
X-Gm-Message-State: ALoCoQnnT5XR2WzJyaDMk0ZtGPJbGJ+HGV9+e3BVpsAuSBaM7+evtF8bDD3h0gOE+kHqnZF6uXgV
X-Received: by 10.66.242.171 with SMTP id wr11mr21567507pac.111.1402586896547;
        Thu, 12 Jun 2014 08:28:16 -0700 (PDT)
Received: from localhost (50-76-60-73-ip-static.hfc.comcastbusiness.net. [50.76.60.73])
        by mx.google.com with ESMTPSA id ir10sm80996154pbc.59.2014.06.12.08.28.15
        for <multiple recipients>
        (version=TLSv1.2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Thu, 12 Jun 2014 08:28:15 -0700 (PDT)
From: Andy Lutomirski <luto@amacapital.net>
To: "H. Peter Anvin" <hpa@zytor.com>, Josh Boyer <jwboyer@fedoraproject.org>
Cc: Michal Marek <mmarek@suse.cz>, linux-kbuild@vger.kernel.org,
        "Linux-Kernel@Vger. Kernel. Org" <linux-kernel@vger.kernel.org>,
        <sam@ravnborg.org>, Andy Lutomirski <luto@amacapital.net>
Subject: [PATCH v3] x86,vdso: Fix vdso_install
Date: Thu, 12 Jun 2014 08:28:10 -0700
Message-Id: <b10299edd8ba98d17e07dafcd895b8ecf4d99eff.1402586707.git.luto@amacapital.net>
X-Mailer: git-send-email 1.9.3
In-Reply-To: <CA+5PVA5rP4cF14-ckD4K4MgBNtfHB+xPg3whEL26JkxegE7qpg@mail.gmail.com>
References: <CA+5PVA5rP4cF14-ckD4K4MgBNtfHB+xPg3whEL26JkxegE7qpg@mail.gmail.com>
X-RedHat-Spam-Score: -2.7  (BAYES_00,DCC_REPUT_13_19,RCVD_IN_DNSWL_LOW,SPF_PASS,URIBL_BLOCKED)
X-Scanned-By: MIMEDefang 2.68 on 10.5.110.16

The filenames are different now.  Inspired by a patch from
Sam Ravnborg.

Acked-by: Sam Ravnborg <sam@ravnborg.org>
Reported-by: Josh Boyer <jwboyer@fedoraproject.org>
Signed-off-by: Andy Lutomirski <luto@amacapital.net>
---

I kept Sam's ack from v2, since v3 has only a trivial change.

Changes from v1: Remove core kbuild change
Changes from v2: Name the result .so files again, not .so.dbg

 arch/x86/vdso/Makefile | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/x86/vdso/Makefile b/arch/x86/vdso/Makefile
index 9769df0..845b45e 100644
--- a/arch/x86/vdso/Makefile
+++ b/arch/x86/vdso/Makefile
@@ -9,11 +9,6 @@ VDSOX32-$(CONFIG_X86_X32_ABI)	:= y
 VDSO32-$(CONFIG_X86_32)		:= y
 VDSO32-$(CONFIG_COMPAT)		:= y
 
-vdso-install-$(VDSO64-y)	+= vdso.so
-vdso-install-$(VDSOX32-y)	+= vdsox32.so
-vdso-install-$(VDSO32-y)	+= $(vdso32-images)
-
-
 # files to link into the vdso
 vobjs-y := vdso-note.o vclock_gettime.o vgetcpu.o
 
@@ -176,15 +171,20 @@ VDSO_LDFLAGS = -fPIC -shared $(call cc-ldoption, -Wl$(comma)--hash-style=sysv) \
 GCOV_PROFILE := n
 
 #
-# Install the unstripped copy of vdso*.so listed in $(vdso-install-y).
+# Install the unstripped copies of vdso*.so.
 #
-quiet_cmd_vdso_install = INSTALL $@
-      cmd_vdso_install = cp $(obj)/$@.dbg $(MODLIB)/vdso/$@
-$(vdso-install-y): %.so: $(obj)/%.so.dbg FORCE
+quiet_cmd_vdso_install = INSTALL $(@:install_%=%)
+      cmd_vdso_install = cp $< $(MODLIB)/vdso/$(@:install_%=%)
+
+vdso_img_insttargets := $(vdso_img_sodbg:%.dbg=install_%)
+
+$(MODLIB)/vdso: FORCE
 	@mkdir -p $(MODLIB)/vdso
+
+$(vdso_img_insttargets): install_%: $(obj)/%.dbg $(MODLIB)/vdso FORCE
 	$(call cmd,vdso_install)
 
-PHONY += vdso_install $(vdso-install-y)
-vdso_install: $(vdso-install-y)
+PHONY += vdso_install $(vdso_img_insttargets)
+vdso_install: $(vdso_img_insttargets) FORCE
 
 clean-files := vdso32-syscall* vdso32-sysenter* vdso32-int80*
-- 
1.9.3

